### PR TITLE
fix: optimize bubble replacement logic with silent replace support

### PR DESCRIPTION
**Pull request by 52cyb**
1. Refactored BubbleModel API: replaced `isReplaceBubble()` and `replaceBubble(BubbleItem*)` with cleaner
`replaceBubbleIndex(NotifyEntity)`, `updateBubbleInPlace()`, and `replaceBubble(int, BubbleItem*)` methods
2. Added support for "silent replace" via `x-deepin-SilentReplace` hint: when two bubbles match and the hint is true, the existing bubble is updated in-place using `dataChanged` rather than full remove+insert
3. Changed `replaceBubbleIndex` to accept `NotifyEntity` instead of `BubbleItem*`, enabling replacement logic before creating a new `BubbleItem`
4. `replaceBubble` now handles resource cleanup by calling `deleteLater()` on both old and new bubbles as needed
5. Removed the no longer needed `isReplaceBubble()` method, simplifying the public interface

Log: Refactored bubble replacement mechanism; added silent in-place update for performance

Influence:
1. Test notification replacement with and without `x-deepin- SilentReplace` hint
2. Verify that in-place update works for silent replacements (no visual artifacts)
3. Test replacement with different app names and bubble IDs to confirm no false matches
4. Verify memory management: old bubbles are properly deleted via `deleteLater()`
5. Test boundary cases: invalid replaceIndex, null bubble, concurrent notifications
6. Verify that `dataChanged` signal correctly triggers view updates for non-silent replacements

refactor: 优化气泡替换逻辑，支持静默替换

1. 重构 BubbleModel API：将 `isReplaceBubble()` 和 `replaceBubble(BubbleItem*)` 替换为更清晰的
`replaceBubbleIndex(NotifyEntity)`、`updateBubbleInPlace()` 和 `replaceBubble(int, BubbleItem*)`
2. 通过 `x-deepin-SilentReplace` 提示增加"静默替换"支持：当气泡匹配且提 示为 true 时，使用 `dataChanged` 原地更新现有气泡，避免完整的移除-插入
操作
3. 将 `replaceBubbleIndex` 改为接受 `NotifyEntity` 而非 `BubbleItem*`， 允许在创建 `BubbleItem` 之前执行替换逻辑
4. `replaceBubble` 现在通过调用新旧气泡的 `deleteLater()` 来处理资源清理
5. 移除不再需要的 `isReplaceBubble()` 方法，简化公开接口

Log: 重构气泡替换机制；增加静默原地更新以提升性能

Influence:
1. 测试带或不带 `x-deepin-SilentReplace` 提示的通知替换
2. 验证静默替换时原地更新正常工作（无视觉异常）
3. 测试不同应用名和气泡 ID 的替换，确认无错误匹配
4. 验证内存管理：旧气泡通过 `deleteLater()` 正确释放
5. 测试边界情况：无效的 replaceIndex、空指针、并发通知
6. 验证对于非静默替换，`dataChanged` 信号正确触发视图更新

PMS: BUG-367205

## Summary by Sourcery

Refactor notification bubble replacement to support index-based replacement and silent in-place updates driven by NotifyEntity metadata.

New Features:
- Add support for silent in-place bubble updates when the x-deepin-SilentReplace hint is set, using dataChanged instead of remove/insert.

Enhancements:
- Expose a new BubbleModel API that operates on NotifyEntity and explicit replace indexes, simplifying and clarifying replacement handling.
- Improve safety and robustness of bubble replacement by validating indexes and centralizing deletion of old BubbleItem instances.

### CODE REVIEW

File: panels/notification/bubble/bubblemodel.cpp [hunk 1]
```diff
@@ -84,15 +84,51 @@ void BubbleModel::insertBubble(BubbleItem *bubble)
...
+    return -1;
+}
+
+void BubbleModel::updateBubbleInPlace(int replaceIndex, const NotifyEntity &entity)
+{
+    if (replaceIndex < 0 || replaceIndex >= m_bubbles.size() || !entity.isReplace())
+        return;
+
+    // dataChanged 通知视图刷新
+    // 前提：内容变更不会引起气泡高度变化，否则布局不刷新（应改用 replaceBubble）。
```
Comment (18202781743) [inline panels/notification/bubble/bubblemodel.cpp:112]: 这个调用方怎么知道内容不会引起高度变化呢？
